src/kademlia/routing_table.cpp

```diff
@@ -1125,8 +1139,7 @@ void routing_table::add_router_node(udp::endpoint router)
 // pinged == false)
 void routing_table::heard_about(node_id const& id, udp::endpoint const& ep)
 {
-	// only when the node_id pass the verification, add it to routing table.
-	if (m_settings.enforce_node_id && !verify_id(id, ep.address())) return;
+	if (!verify_node_address(m_settings, id, ep.address())) return;
```

Comment:
I think the IP check may have to happen a lot earlier. I don't think it's reasonable to react to any invalid dht messages at all (this patch only avoids remembering them in the routing table)